src/core/datagram.c

```diff
@@ -421,7 +421,13 @@ QuicDatagramSendFlush(
         CXPLAT_DBG_ASSERT(!(SendRequest->Flags & QUIC_SEND_FLAG_BUFFERED));
         CXPLAT_TEL_ASSERT(Datagram->SendEnabled);
 
-        if (SendRequest->TotalLength > (uint64_t)Datagram->MaxSendLength || QuicConnIsClosed(Connection)) {
+        // Cancels the sending of a Datagram if:
+        // The length we want to send > the length we can send
+        // The connection is closed
+        // The sending mode requires to send immediately (no datagram in queue)
```

Comment:
Just because there is another datagram in the queue doesn't mean we should cancel this one, IMO. This logic should be moved to `QuicSendFlush`, to cancel any datagrams that can't be sent the next time we send out packets.

Reply:
Actually, I printed out the values we had in the Datagram object to check what changes when we do or do not have congestion, and that was the only thing that changed (from NULL to a value) that is why I've done that check

Reply:
I think I understand what you meant sorry, you want to extend it to Quic packets and not only Quic Datagrams ?